lambda-runtime/src/AWS/Lambda/Client.hs

```diff
@@ -0,0 +1,175 @@
+module AWS.Lambda.Client
+  ( HTTPConfig
+  , InternalLambdaClientError (..)
+  , LambdaClient
+  , LambdaResponse (..)
+  , getHttpConfig
+  , getNextLambdaResponse
+  , sendBootError
+  , sendEvent
+  )
+where
+
+import AWS.Prelude
+import Control.Exception (displayException)
+import Control.Monad.Except (ExceptT(..), liftEither, throwError, withExceptT)
+import System.Environment (lookupEnv)
+
+import qualified Data.Aeson          as JSON
+import qualified Data.ByteString     as BS
+import qualified Network.HTTP.Client as HTTP
+import qualified Network.HTTP.Types  as HTTP
+
+data HTTPConfig = HTTPConfig
+  { request :: HTTP.Request
+  , manager :: HTTP.Manager
+  }
+
+data LambdaResponse = LambdaResponse
+  { event     :: JSON.Value
+  , requestId :: RequestId
+  }
+  deriving stock Show
+
+newtype RequestId = RequestId Text
+  deriving stock (Show)
+
+instance Conversion BS.ByteString RequestId where
+  convert (RequestId requestId) = encodeUtf8 requestId
+
+data InternalLambdaClientError
+  = LambdaContextDecodeError Text
+  | MissingLambdaRunTimeApi
+  | InvalidLambdaRunTimeApi Text
+  | ConnectionError HTTP.HttpException
+  | ResponseBodyDecodeFailure Text
+  | UnSuccessfulEventRetrieval HTTP.Status Text
+  | UnSuccessfulEventSending HTTP.Status Text
+  | PayloadTooLarge Text
+  deriving stock Show
+
+instance Exception InternalLambdaClientError
+
+type LambdaClient = ExceptT InternalLambdaClientError IO
+
+getHttpConfig :: LambdaClient HTTPConfig
+getHttpConfig = do
+  awsLambdaRuntimeApi <- ExceptT $ maybeToEither MissingLambdaRunTimeApi
+    <$> lookupEnv "AWS_LAMBDA_RUNTIME_API"
+
+  req <- withExceptT (InvalidLambdaRunTimeApi . showc @Text)
+    . ExceptT
+    . try @_ @HTTP.HttpException
+    $ HTTP.parseRequest ("http://" <> awsLambdaRuntimeApi)
+
+  manager <- liftIO $ HTTP.newManager
+    $ HTTP.managerSetProxy HTTP.noProxy
+    $ HTTP.defaultManagerSettings
+        { HTTP.managerResponseTimeout     = HTTP.responseTimeoutNone
```

Comment:
@epicallan thanks for getting that one right, you recall the `hal` case ;)

Reply:
> thanks for getting that one right, you recall the hal case ;)

I wish I did, can't seem to recall.

Reply:
The author of HAL uses a timeout here, but as the container is frozen when there is no event the clock may jump forward immediately whne the container is unfrozen running into the timeout before reading the first event after unfreeze.

NOT timing out removes this source of event drop errors.